### PR TITLE
editor: Add revert file action to command palette

### PR DESCRIPTION
**Pull request by CharlesChen0823**
Add revert file action to command palette.

Release Notes:

- N/A

### CODE REVIEW

File: crates/editor/src/hunk_diff.rs [hunk 1]
```diff
@@ -139,31 +139,12 @@ impl Editor {
                             }
                         }
                     })
-                    .entry("Revert File", None, {
+                    .entry("Revert File", Some(RevertFile.boxed_clone()), {
```
Comment (SomeoneToIgnore) [inline crates/editor/src/hunk_diff.rs:142]: Would `.action(` work here? That would be a shorter declaration then.

https://github.com/zed-industries/zed/blob/207f1aea4772c3605f5f609dfce6b80375b63f18/crates/ui/src/components/context_menu.rs#L156

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks good, let's try one small style fix and merge this.

File: crates/editor/src/hunk_diff.rs (hunk 1 above)
Comment (CharlesChen0823) [reply crates/editor/src/hunk_diff.rs]: Done

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you.

**Pull request merged**